File: programs/blocking-connection/blocking-connection-server.java
import java.io.*;
import java.net.*;

public class TCPServer {
    public static void main(String[] args) {
        int port = 666;
        try (ServerSocket serverSocket = new ServerSocket(port)) {
            System.out.println("Server listening in port: " + port);

            Socket socket = serverSocket.accept();
            System.out.println("Connected from: " + socket.getInetAddress());

            BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));

            PrintWriter out = new PrintWriter(socket.getOutputStream(), true);

            String inputLine;
            while ((inputLine = in.readLine()) != null) {
                System.out.println("Received: " + inputLine);
                out.println(inputLine);
            }
            socket.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: programs/blocking-connection/blocking-connection-server.py
import socket

HOST = '127.0.0.1'
PORT = 666

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
    s.bind((HOST, PORT))
    s.listen()

    print(f"Server listening on {HOST}:{PORT}...")

    conn, addr = s.accept()

    with conn:
        print(f"Connection established from {addr}")
        while True:
            data = conn.recv(1024)
            if not data:
                break
            print("Received data:", data.decode())
            conn.sendall(data)
        print("Connection closed.")
